src/OrleansRuntime/GrainTypeManager/GrainTypeManager.cs

```diff
@@ -151,24 +149,27 @@ internal IReadOnlyList<SiloAddress> GetSupportedSilos(int typeCode)
             return supportedSilosByTypeCode[typeCode];
         }
 
-        internal IReadOnlyList<SiloAddress> GetSupportedSilos(int typeCode, int ifaceId, ushort version)
+        internal IReadOnlyList<SiloAddress> GetSupportedSilos(int typeCode, int ifaceId, IReadOnlyList<ushort> versions)
         {
-            var key = Tuple.Create(typeCode, ifaceId, version);
-            return supportedSilosCache.GetOrAdd(
-                key,
-                tuple =>
-                {
-                    var silosWithTypeCode = supportedSilosByTypeCode[typeCode];
-                    return supportedSilosByInterface[ifaceId][version].Intersect(silosWithTypeCode).ToList();
-                });
+            var result = new List<SiloAddress>();
+            foreach (var version in versions)
```

Comment:
I agree but I would argue that we call this method only once per type/iface/version since it is cached at an upper level. I prefer to use a strict param since no modification should be done to that list.